### PR TITLE
[subset] Experimental: store a list of (old, new) gid pairs for quick iteration.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: src/hb-ot-hmtx-table.hh [hunk 1]
```diff
@@ -211,20 +211,22 @@ struct hmtxvmtx
...
-		    return hb_pair (0u, 0);
-		  int lsb = 0;
-		  if (!_mtx.get_leading_bearing_without_var_unscaled (old_gid, &lsb))
-		    (void) _glyf_get_leading_bearing_without_var_unscaled (c->plan->source, old_gid, !T::is_horizontal, &lsb);
-		  return hb_pair (_mtx.get_advance_without_var_unscaled (old_gid), +lsb);
-		}
-		return mtx_map->get (_);
+                hb_codepoint_t old_gid = p.first;
+                hb_codepoint_t new_gid = p.second;
+                if (new_gid == -1u)
```
Comment (behdad) [inline src/hb-ot-hmtx-table.hh:219]: `HB_MAP_VALUE_INVALID`.

**Comment by garretrieger** [comment]
Initial benchmarks look good:

BM_subset/subset_glyphs/Mplus1p-Regular.ttf/10_median                             -0.1433         -0.1433             0             0             0             0
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/64_median                             -0.1800         -0.1798             0             0             0             0
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/512_median                            -0.1196         -0.1192             1             1             1             1
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/4096_median                           -0.0808         -0.0807             3             3             3             3
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/10000_median                          -0.0104         -0.0104             8             7             8             7
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/nohinting/10_median                   -0.1574         -0.1576             0             0             0             0
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/nohinting/64_median                   -0.1435         -0.1437             0             0             0             0
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/nohinting/512_median                  -0.0903         -0.0902             1             1             1             1
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/nohinting/4096_median                 -0.0598         -0.0597             3             3             3             3
BM_subset/subset_glyphs/Mplus1p-Regular.ttf/nohinting/10000_median                -0.0247         -0.0248             8             7             8             7

**Comment by garretrieger** [comment]
This is not entirely correct as it changes the output to be ordered by old gid, which may not be right if a custom glyph mapping is being used. Need to switch the map sorting from new -> old

**Comment by behdad** [comment]
Alternative approach: https://github.com/harfbuzz/harfbuzz/pull/4254

**Comment by garretrieger** [comment]
> Alternative approach: #4254

Thanks, let's use that one. I'll close this.

**Pull request closed without merging**